code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -29,9 +33,16 @@
 class QuestionAnswerTool(AnsweringToolBase):
     def __init__(self) -> None:
         self.name = "QuestionAnswer"
-        self.vector_store = AzureSearchHelper().get_vector_store()
-        self.verbose = True
         self.env_helper = EnvHelper()
+        self.llm_helper = LLMHelper()
+        if self.env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
```

Comment:
We have this code block a few times in the codebase, wondering whether a factory method would be better would also help testing